### DISCUSSION TITLE
CLI Native OS Builds

### DISCUSSION THREAD

**Discussion by trymeouteh** [parcel-bundler/lightningcss#732] [Ideas]
If Lightning CSS CLI is written in Rust, could it be compiled into CLI executable for Windows, MacOS and Linux?

I would prefer to install Lightning CSS as a native executable on my machine.

Would be nice to see the option to install Lightning CLI in the following ways...

- Windows
   - Winget
   - Scoop
- MacOS
   - Homebrew
 - Linux
   - Debian/Ubuntu Repo
   - Arch Repo
   - Nix

**Comment by mischnic** [comment]
If you install the npm package `lightningcss-cli`, you do actually get a native Rust CLI. 
So you absolutely can do this, but not via the system package manager:
> I would prefer to install Lightning CSS as a native executable on my machine.

**Reply by trymeouteh** [reply]
Is it possible to release the `lightningcss-cli` as a system package, to prevent users from needing to install npm to install the package?

**Reply by devongovett** [reply]
Possible, but it's a lot more to maintain so I'm hesitant to take it on.

You could always just download the tarball from the npm registry directly without needing node or npm, for example using a URL like https://registry.npmjs.org/lightningcss-cli-darwin-arm64/-/lightningcss-cli-darwin-arm64-1.25.0.tgz. That will contain a `lightningcss` executable compiled for the platform you requested (darwin-arm64 in this case).
